kernel/common/enumerable.rb

```diff
@@ -567,86 +567,110 @@ def first(n=undefined)
     nil
   end
 
-  def min
-    min = undefined
-    each do
-      o = Rubinius.single_block_arg
-      if undefined.equal? min
-        min = o
-      else
-        comp = block_given? ? yield(o, min) : o <=> min
-        unless comp
-          raise ArgumentError, "comparison of #{o.class} with #{min} failed"
-        end
-
-        if Comparable.compare_int(comp) < 0
+  def min(n=nil)
+    if n.nil?
```

Comment:
I can't really read it from the diff, but is this method supposed to return `nil` if `n` is also `nil`? In that case this is much better:

    return nil if n.nil?